packages/admin-portal/src/components/pages/apps-management/check-aws-button.tsx

```diff
@@ -25,7 +25,9 @@ export const checkIsAws = (subscriptions: WebhookModel[]): boolean => {
   const filtered = subscriptions.filter((subscription) => {
     const { topicIds, customerIds, active } = subscription
     const hasDataTopics = Boolean(topicIds?.filter((topic: string) => !INSTALL_TOPICS.includes(topic)).length)
-    const hasCustomerSubs = Boolean(customerIds?.filter((id: string) => !FILTERED_CLIENT_IDS.includes(id)).length)
+    const hasCustomerSubs = Boolean(
+      !customerIds?.length || customerIds?.filter((id: string) => !FILTERED_CLIENT_IDS.includes(id)).length,
+    )
```

Comment:
this is hurting my brain a bit. Probably me as it's currently mush from all those aws issues. But here you're checking if length exists? `length >= 1` as `0` would result in truthy. and >= 1 would be false? If you just want to check if customerIds is not an array `!Array.isArray(customerIds)` would make more sense.

The second part is filtering the array and making sure the ids exist in the const? and checking there is at least one that exists? If so there's a function called `some` which might make more sense https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
Some will return true if there is at least one present (not sure if it breaks once found) `customerIds.some(id => FILTERED_CLIENT_IDS.includes(id))`

This is all providing I've read this correctly?

Reply:
update: I didn't read it correctly

Reply:
It's a bit weird - basically if the array is empty, the webhooks apply to all customers. If it's not empty, I want to filter it to see if it has length when SBOX has been removed. Worth looking at the test cases as it's a bit more obvious what's going on...